### PR TITLE
Improve the result of GetCamelCasedSuggestion

### PR DESCRIPTION
**Pull request by Porges**
(Moved from https://github.com/Azure/autorest/pull/2548)

When run on our service's definition, `GetCamelCasedSuggestion` returned:

* `nodeAgentSkUid` for `nodeAgentSKUId`
* `inboundNaTpools` for `inboundNATPools`

These are patently silly. This change improves them to:

* `nodeAgentSkuId` for `nodeAgentSKUId`
* `inboundNatPools` for `inboundNATPools`

### CODE REVIEW

**Comment by olydis** [comment]
@dsgouda PTAL

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs [hunk 1]
```diff
@@ -365,33 +365,14 @@ private static IEnumerable<Operation> SelectOperationsFromPaths(string id, Dicti
...
-                    firstUpper = true;
-                    if (char.IsUpper(sb[i]))
-                    {
-                        sb[i] = sb[i].ToString().ToLower()[0];
-                    }
-                }
-            }
-            return sb.ToString();
-        }
+            => Regex.Replace(name, @"(?<=\p{Lu})\p{Lu}+(?=\p{Lu}|\z)", s => s.Value.ToLowerInvariant());
```
Comment (dsgouda) [inline src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: I'm all for regexes but even I can't comprehend this on the first go. I'd keep the dirty logic as is since this is bound to evolve further. @veronicagg thoughts?

Comment (olydis) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: note that the logic is *changed* to actually alter the rule, so there's not much to "keep". But I agree, there may be value in rewriting this Regex as code (preferably not dirty, but readable 😄 ) for future maintainability

Comment (dsgouda) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: Ahh, did not notice that, but yes, still see value in turning the regex into code.

Comment (veronicagg) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: Yeah, the reason I haven't reviewed this PR is because I had to set time aside to study the regex :) if we can put it in simpler terms it'd be great, if not, may be an explanation of the regex in comments would help. The current logic suggesting a string is not ideal, but that's the reason we kept it as a suggestion, it was a way to "show" the issue and a compliant option more than having people take the suggestion.

Comment (Porges) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: It's _quite_ clunky if converted to code but I'll comment the regex to explain what it's doing

Comment (Porges) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: ... done that, and added a test for expected behaviour.

**Comment by Porges** [comment]
Thanks!
